platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/GeoJsonOptions.java

```diff
@@ -109,4 +109,33 @@ public GeoJsonOptions withClusterRadius(int clusterRadius) {
     this.put("clusterRadius", clusterRadius);
     return this;
   }
+
+  /**
+   * An object defining custom properties on the generated clusters if clustering is enabled,
+   * aggregating values from clustered points. Has the form {"property_name": [operator, [map_expression]]} or
+   * {"property_name": [[operator, accumulated, expression], [map_expression]]}
+   *
+   * @param propertyName name of the property
+   * @param operator operator is any expression function that accepts at least 2 operands (e.g. "+" or "max") - it
+   *                 accumulates the property value from clusters/points the cluster contains. It can either be a
+   *                 string with single operator, or be an object array converted from valid expression
+   * @param mapExpr map expression produces the value of a single point, it shall be an object array, which is
+   *                converted from a valid expression
+   * @return the current instance for chaining
+   */
+  @NonNull
+  public GeoJsonOptions withClusterProperty(String propertyName, Object operator, Object mapExpr) {
+    if (this.containsKey("clusterProperties")) {
+      HashMap<String, Object[]> properties = (HashMap<String, Object[]>)(this.get("clusterProperties"));
+      properties.put(propertyName, new Object[]{operator, mapExpr});
+      properties.put(propertyName, new Object[]{operator, mapExpr});
+      this.put("clusterProperties", properties);
+    } else {
+      HashMap<String, Object[]> properties = new HashMap<String, Object[]>();
+      properties.put(propertyName, new Object[]{operator, mapExpr});
+      properties.put(propertyName, new Object[]{operator, mapExpr});
+      this.put("clusterProperties", properties);
+    }
```

Comment:
there is some duplicate code in above if block, this could be simplified either with a ternary operator:
```java
    HashMap<String, Object[]> properties = containsKey("clusterProperties")
      ? (HashMap<String, Object[]>) get("clusterProperties") : new HashMap<String, Object[]>();
    properties.put(propertyName, new Object[]{operator, mapExpr});
    properties.put(propertyName, new Object[]{operator, mapExpr});
    put("clusterProperties", properties);
    return this;
```
or the if could be reworked to:

```java
    HashMap<String, Object[]> properties;
    if (containsKey("clusterProperties")) {
      properties = (HashMap<String, Object[]>) get("clusterProperties");
    } else {
      properties = new HashMap<>();
    }
    properties.put(propertyName, new Object[]{operator, mapExpr});
    properties.put(propertyName, new Object[]{operator, mapExpr});
    put("clusterProperties", properties);
    return this; 
```


note: in above codeblocks I removed references to this as they are implicit, additionally removed the generic type declaration if initialization in in-line (eg. `properties = new HashMap<>();`